### PR TITLE
Codec adapter fixes

### PR DESCRIPTION
**Pull request by keyonjie**
minor fixes for dummy codec_adapter.

### CODE REVIEW

File: src/audio/codec_adapter/codec_adapter.c [hunk 1]
```diff
@@ -139,8 +139,8 @@ static int load_setup_config(struct comp_dev *dev, void *cfg, uint32_t size)
 			 (uintptr_t)cfg, size);
 		ret = -EINVAL;
 		goto end;
-	} else if (size <= sizeof(struct ca_config)) {
-		comp_err(dev, "load_setup_config(): no codec config available.");
+	} else if (size < sizeof(struct ca_config)) {
```
Comment (lgirdwood) [inline src/audio/codec_adapter/codec_adapter.c:142]: Commit message "dummy" -> "passthrough".

**Review by lgirdwood — CHANGES_REQUESTED** [review]
Needs inclusive language.
@deb-intel @intelkevinputnam can we add the woke checker yet as a GH action for source code PRs ?

**Review by dbaluta — APPROVED** [review]

File: src/audio/codec_adapter/codec_adapter.c [hunk 2]
```diff
@@ -154,13 +154,15 @@ static int load_setup_config(struct comp_dev *dev, void *cfg, uint32_t size)
...
 	}
 	/* Copy codec specific part */
-	lib_cfg = (char *)cfg + sizeof(struct ca_config);
 	lib_cfg_size = size - sizeof(struct ca_config);
-	ret = codec_load_config(dev, lib_cfg, lib_cfg_size, CODEC_CFG_SETUP);
-	if (ret) {
-		comp_err(dev, "load_setup_config(): %d: failed to load setup config for codec id %x",
-			 ret, cd->ca_config.codec_id);
-		goto end;
+	if (lib_cfg_size > 0) {
```
Comment (ranj063) [inline src/audio/codec_adapter/codec_adapter.c:158]: Just if(lib_cfg_size) should do?

File: src/audio/codec_adapter/codec_adapter.c [hunk 3]
```diff
@@ -154,13 +154,15 @@ static int load_setup_config(struct comp_dev *dev, void *cfg, uint32_t size)
...
+		lib_cfg = (char *)cfg + sizeof(struct ca_config);
+		ret = codec_load_config(dev, lib_cfg, lib_cfg_size, CODEC_CFG_SETUP);
+		if (ret) {
+			comp_err(dev, "load_setup_config(): %d: failed to load setup config for codec id %x",
+				 ret, cd->ca_config.codec_id);
+			goto end;
```
Comment (ranj063) [inline src/audio/codec_adapter/codec_adapter.c:164]: I'd just return ret here. Why go-to just to do the same thing

File: src/audio/codec_adapter/codec_adapter.c [hunk 4]
```diff
@@ -154,13 +154,15 @@ static int load_setup_config(struct comp_dev *dev, void *cfg, uint32_t size)
...
+		}
```
Comment (ranj063) [inline src/audio/codec_adapter/codec_adapter.c:165]: In fact, we should remove all goto end and replace with return ret. And return 0 at the end

Comment (keyonjie) [reply src/audio/codec_adapter/codec_adapter.c:165]: > In fact, we should remove all goto end and replace with return ret. And return 0 at the end

That should be another separated refinement if needed, this PR here only aim to fix the 0 codec config size issue.

File: src/audio/codec_adapter/codec_adapter.c (hunk 2 above)
Comment (keyonjie) [reply src/audio/codec_adapter/codec_adapter.c]: > Just if(lib_cfg_size) should do?

done.

File: src/audio/codec_adapter/codec_adapter.c (hunk 4 above)
Comment (ranj063) [reply src/audio/codec_adapter/codec_adapter.c:165]: ok lets do that in a separate PR

**Review by ranj063 — APPROVED** [review]

**Review by lgirdwood — APPROVED** [review]

**Pull request merged**